### ISSUE TITLE
line breaks are missing from search results

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud.solr#67]
I have this one-cell notebook:

```r
require(rcloud.web)
rcw.result(
   run=function() rcw.append("#test", "<p>Hello from R!"),
   body="<div id=test>This is my nice HTML...</div>")
```

If I search for a term from the first line, I get bits of the next line:

![image](https://user-images.githubusercontent.com/1366709/33561719-6cd7c9f8-d8e1-11e7-8219-8446449cbde7.png)

If I search for a term from a later line, I get bits and pieces of the previous line and matching line:

![image](https://user-images.githubusercontent.com/1366709/33561804-a813b8ce-d8e1-11e7-83eb-3dcbb74b625a.png)

Part of the problem here is that special HTML characters need to be escaped. But I'd expect to see either just the matching line, or the complete surrounding lines with line breaks.

**Comment by dougmet** [comment]
Yes this should be fixable. The problem begins because we're keeping a certain number of characters around the hit. However, should be able to trim it at line breaks.

Some of the more obscure code from the previous version was trying to handle this problem. Would be nice if we can find something with a similar effect but not so heavy on R code.

**Comment by gordonwoodhull** [comment]
Ah yes, I can see why this problem could cause less experienced programmers to generate a lot of code. 

If there's an R package that can help finding which line a character is on, and breaking into lines, feel free to use it. We tend to assume that an RCloud install will have most of CRAN anyway.
